### PR TITLE
Stripe appears

### PR DESCRIPTION
**Pull request by Ru-T**
## What Changed?
* Add stripe gem and configuration
* Add javascript for processing payment form
* Add controller details in create action for creation of customer and of charge
* Add card_token to orders table

## Why? 
* So folks can PAY!

### CODE REVIEW

File: app/assets/javascripts/order.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    return false;
+  });
+});
+
+stripeResponseHandler = function(status, response) {
+   var $form = $("#new_order");
+
+   if (response.error) {
+     $form.find('.payment-errors').text(response.error.message);
+     $form.find("button").prop("disabled", false);
```
Comment (jahammo2) [inline app/assets/javascripts/order.js]: No biggie but be sure to keep your quotes (single vs double) consistent

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -10,6 +10,22 @@ def create
...
     if @order.save
       @order.purchase_line_items
+
+      customer = Stripe::Customer.create(
+        email:  @order.user.email,
+        source: params[:stripeToken]
+      )
+      @order.card_token = customer.id
+
+      amount = @order.total
```
Comment (jahammo2) [inline app/controllers/orders_controller.rb]: If I'm not mistaken, you shouldn't need to continue calling `order` as an instance variable (`@order`). After you define it on line 10, you can just call `order` afterwards. I still don't understand why but I've had it drilled in my head that
screen shot 2016-02-04 at 5 39 13 pm

File: features/purchase.feature [hunk 3]
```diff
@@ -1,13 +1,58 @@
...
+    When I click "Purchase"
+    Then I am asked for my shipping address
+    When I enter my shipping address
+    Then I am asked if I want to use my already saved credit card
+    When I confirm using my saved credit card
+    Then I am asked to review the order total
+    When I click "Confirm"
+    Then I am shown the order summary
+    And my credit card is saved for future purchases
+    And I am emailed an order invoice containing the books details, quantity, subtotal, and order total
```
Comment (jahammo2) [inline features/purchase.feature]: Excellent tests!

File: features/step_definitions/purchase_steps.rb [hunk 4]
```diff
@@ -28,7 +38,78 @@
...
+Then(/^I am shown the order summary$/) do
+  expect(page).to have_content("Order Summary")
+end
+
+Then(/^my credit card is saved for future purchases$/) do
+  expect(@user.card_token).to_not be_nil
+end
+
+Then(/^I am emailed an order invoice containing the books details, quantity, subtotal, and order total$/) do
+  ActionMailer::Base.deliveries.last.body.to_include("receipt")
```
Comment (jahammo2) [inline features/step_definitions/purchase_steps.rb]: Daaaaammmmmnnnn. Testing that ActionMailer.

**Comment by jahammo2** [comment]
This is mad gucci. Just change out those instance variables (as long as it doesn't break anything. If it does, ignore everything I said) and :shipit:

File: features/step_definitions/purchase_steps.rb (hunk 4 above)
Comment (Ru-T) [inline features/step_definitions/purchase_steps.rb]: You know it.

File: features/purchase.feature (hunk 3 above)
Comment (Ru-T) [inline features/purchase.feature]: :tada:

File: app/controllers/orders_controller.rb (hunk 2 above)
Comment (Ru-T) [inline app/controllers/orders_controller.rb]: undefined local variable or method `order' for #<OrdersController:0x007f967c894d28> Did you mean? @order

:(

Comment (jahammo2) [inline app/controllers/orders_controller.rb]: Oh no, it's okay! I wasn't sure if this was even possible. But carry on as is!

**Pull request merged**